spec/saml_response_spec.rb

```diff
@@ -0,0 +1,104 @@
+require_relative "spec_helper"
+require_relative "../lib/fake_idp/saml_response"
+require "ruby-saml"
+require "securerandom"
+
+RSpec.describe FakeIdp::SamlResponse do
+  before do
+    FakeIdp.configure do |config|
+      config.callback_url = "http://localhost.dev:3000/auth/saml/devidp/callback"
+      config.sso_uid = "12345"
+      config.username = "bobthessouser"
+      config.name_id = "bobthessouser@example.com"
+      config.first_name = "Reid"
+      config.last_name = "Smith"
+      configuration.email = "reid@msn.com"
+      config.idp_certificate = fake_public_certificate
+      config.idp_secret_key = fake_private_key
+      config.algorithm = :sha1
+    end
+  end
+
+  let(:configuration) { FakeIdp.configuration }
+
+  it "generates a valid SAML response" do
+    settings = OneLogin::RubySaml::Settings.new(
+      name_identifier_format: "urn:oasis:names:tc:SAML:1.1:nameid-format:unidentified",
```

Comment:
Would it make sense for `name_identifier_format` to be a potential config variable for `FakeIdp`, similar to `idp_certificate`, `algorithm`, etc?

Reply:
Yes, that might be helpful but on second thought, it may not be necessary to explicitly pass the `name_identifier_format` in the settings here. I'll investigate this a bit